kong/plugins/grpc-web/handler.lua

```diff
@@ -34,6 +37,26 @@ local CORS_HEADERS = {
   ["Access-Control-Allow-Headers"] = "content-type,x-grpc-web,x-user-agent",
 }
 
+local function get_body()
+  local body, err = kong_request_get_raw_body()
+  if body then
+    return body
+  end
+
+  -- if body_file is not nil, the error of get_raw_body is expected
+  -- otherwise return the error
+  local body_file = ngx_req_get_body_file()
+  assert(body_file, err)
+
+  warn("client_body_buffer_size exceeded and reading the request from disk. Please consider increasing the value.")
+
+  local file = assert(io_open(body_file, "rb"))
+  body = assert(file:read("*a"))
```

Comment:
We should be careful with such blocking file io operations. If the file is large, it will block the whole nginx event loop significantly.

Reply:
That's the reason for emitting the warning message. And there's no better solution: we are not using a streaming JSON parser and the `pb_unframe`.

Reply:
A possible solution is to read with limited length and yield after every read.

Reply:
Yeah, this is a possible approach, or perhaps throwing it to another OS thread to avoid blocking the main worker thread.

Reply:
A good candidate for:
https://github.com/openresty/lua-nginx-module#ngxrun_worker_thread

But that needs an addition to nginx.conf, such as:
```
thread_pool default threads=32 max_queue=65536;
```